language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/BallerinaTextDocumentService.java

```diff
@@ -495,9 +495,14 @@ void setClientCapabilities(LSClientCapabilities clientCapabilities) {
 
             Path compilationPath = getUntitledFilePath(docSymbolFilePath.toString()).orElse(docSymbolFilePath.get());
             Optional<Lock> lock = docManager.lockFile(compilationPath);
+
+            LSContext codeLensContext = new DocumentServiceOperationContext
+                    .ServiceOperationContextBuilder(LSContextOperation.TXT_CODE_LENS)
+                    .withCommonParams(null, fileUri, docManager)
+                    .build();
+
             try {
-                // Compile source document
-                lenses = CodeLensUtil.compileAndGetCodeLenses(fileUri, docManager);
+                lenses = CodeLensUtil.getCodeLenses(codeLensContext);
                 docManager.setCodeLenses(compilationPath, lenses);
```

Comment:
We will need to get rid of the code lenses caching within the Document Manager. Once the new Semantic API and Project API is integrated let's remove this part

Reply:
Yes agree, only added changes affected by add-doc code action changes.  Let's remove it coming up PRs.